### ISSUE TITLE
DetailsList components not able to assign tabIndex to its components and the onRender rows

### ISSUE THREAD

**Issue by jcdiocadiz** [microsoft/fluentui#18435]
### Environment Information

- **Package version(s)**: latest
- **Browser and OS versions**: any browser

### Issue

The DetailsList Component has no parameter for assigning the tabIndex and when I tried looking at the Developer's Tool to see if the DetailsList component has tabIndex, it has none. As part of the accessibility testing, we need to ensure that each columns and row cells can be tabbed so that NVDA can speak them or write them to speech. I also assigned tabIndex to the onRender() function of the IColumns but it didn't work. It basically assigned -1 as the TabIndex to them.

This is the DetailsList element looks like:
`<div class="ms-DetailsList is-fixed assignment-tbl root-103" data-automationid="DetailsList" data-is-scrollable="false">`

The headers are also by default 0 value as tabIndex:

`<span class="cellTooltip-131"><span id="header210-column1" class="ms-DetailsHeader-cellTitle cellTitle-129" data-is-focusable="true" aria-haspopup="false" tabindex="0"><span id="header210-column1-name" class="ms-DetailsHeader-cellName cellName-130">Name</span></span></span>
`
I tried assigning the tabIndex in onRender() method but it is still -1 as tabIndex.

**_In Code:_**
 let tabIndex = 4;
   const columns: IColumn[] = [
    {
      key: "column1",
      name: "Name",
      fieldName: "name",
      minWidth: 100,
      maxWidth: 250,
      isRowHeader: true,
      isResizable: true,
      data: "string",
      isPadded: true,
      className: "width-22 cursor-pointer",
      onRender: (item: IAssignments) => {
        return (
          <span aria-label={item.name} tabIndex={tabIndex}>
            {item.name}
          </span>
        );
      },
    },...

_**In Browser Developer's Tool:**_
`<div id="row350-0-header" role="rowheader" aria-readonly="true" aria-colindex="2" class="width-22 cursor-pointer is-row-header isRowHeader-147 ms-DetailsRow-cell cell-142 cellPadded-141" data-automationid="DetailsRowCell" data-automation-key="column1" style="width: 294px;"><span aria-label="new assignment" **tabindex="-1"**>new assignment</span></div>`
      
I am not sure if this has been fixed already but the point here is we should be able to add tabIndex to the entire component including the elements underneath for accessibility purposes. 

#### Actual behavior:

Default tabIndex for header as 0. For rows, default tabIndex is -1.

#### Expected behavior:

The  tabIndex should be based on the tabIndex assigned to it. For now, there is no way to assign it in the component level and not working in the onRender() custom row elements.

### Documentation describing expected behavior

Basically, all elements should be accessible via tab for Accessibility purposes.

**Comment by gouttierre** [comment]
@jcdiocadiz  - Thanks for filing this and providing details into the issue, but you’ll need to add a CodePen repro before we can help. You can start from one of the [DetailsList ](https://developer.microsoft.com/en-us/fluentui#/controls/web/detailslist) examples (click “Export to CodePen”) or https://aka.ms/fluentpen.
